qiskit/circuit/add_control.py

```diff
@@ -31,77 +31,30 @@ def add_control(operation, num_ctrl_qubits, label):
             uses num_ctrl_qubits-1 ancillae qubits so returns a gate of size
             num_qubits + 2*num_ctrl_qubits - 1.
     """
+    import qiskit.extensions.standard as standard
+    # this prevents cyclic imports for some gates
+    known_control_map_c1 = {
+        standard.XGate: standard.CnotGate,
+        standard.U1Gate: standard.Cu1Gate,
+        standard.U3Gate: standard.Cu3Gate,
+        standard.CnotGate: standard.ToffoliGate
```

Comment:
does consolidating the files for these gates cause cyclic imports?

Reply:
Just a comprehensive question: Why can the controlled versions of the X, U1 and U3 gates not be put in the same files as the base gates, as done for the other gates? Why does putting the definitions in the same file not resolve the cyclic import here?

Reply:
@ajavadia Yes, consolidating these gates, as was done for other standard gates, causes cyclic import warnings. 

@Cryoris This is generally due to cases where the gate and its controlled version also depend on other gates in their definitions. The choice of gates here used to stop the cyclic import may not be unique or minimal.

Reply:
If the gate imports are moved to runtime imports in the `_define`/`inverse` functions, is a circular import warning still raised?

Reply:
If making the imports at runtime still raises (if this is an instance of https://github.com/PyCQA/pylint/issues/850 ), I think a `pylint-ignore` is a better answer.

Reply:
Previously it seemed `pylint-ignore` was required to be project wide, as described in the pylint issue, but it seemed to work defining it in the module.